src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -268,10 +268,17 @@ def getReplicaInfoForBlocks(self, **kwargs):
             inputDids.append({"scope": kwargs["scope"], "type": "DATASET", "name": block})
 
         resultDict = {}
-        for item in self.cli.list_dataset_replicas_bulk(inputDids):
-            resultDict.setdefault(item['name'], [])
-            if item['state'].upper() == 'AVAILABLE':
-                resultDict[item['name']].append(item['rse'])
+        if kwargs['deep']:
+            for block in blockNames:
```

Comment:
For the sake of avoiding confusion and keeping consistency, I think it would be better to iterate through `inputDids` instead of `blockNames`. What do you think?

Instead of passing `deep=True`, I would suggest to replace it by `deep=kwargs['deep']`.

Reply:
Ok, it makes sense.